src/lib/Microsoft.Health.Fhir.Ingest/Telemetry/FhirExceptionTelemetryProcessor.cs

```diff
@@ -29,14 +33,26 @@ public FhirExceptionTelemetryProcessor()
                 typeof(CorrelationIdNotDefinedException),
                 typeof(InvalidQuantityFhirValueException))
         {
+            _errorMessageService = errorMessageService;
         }
 
-        public override bool HandleException(Exception ex, ITelemetryLogger logger)
+        public override bool HandleException(Exception ex, JToken message, ITelemetryLogger logger)
         {
             EnsureArg.IsNotNull(ex, nameof(ex));
             EnsureArg.IsNotNull(logger, nameof(logger));
 
             var exceptionTypeName = ex.GetType().Name;
+
+            // send to error message service
+            if (_errorMessageService != null)
+            {
+                var errorMessage = new ErrorMessage();
```

Comment:
The processor is defined once, the message is supposed to be per exception so don't think the JToken in the ctor is appropriate. 

Overall I think we might want to refactor this to introduce a context that would wrap the Exception, a reference to what is being processed, and any other data that is relevant.  This context can then be used to generate the error message.

This doesn't need to be available in the Azure Function OSS version. That may allow you more freedom in the implementation.

Reply:
Thanks Nate. Having the mapping in the ErrorMessage class or internally in another class makes sense to me.

@dustinburson we could have an ErrorContext class that wraps an exception and a reference to what is processed with a "ToErrorMessage" method. Is the advantage of that so that we have a strongly typed class that includes exception + payload that we pass around instead of independently passing an exception and a reference to the payload as separate arguments?